pkg/chunk/elastic/es_index_client.go

```diff
@@ -0,0 +1,344 @@
+			panic(err)
+		}
+		if !createIndex.Acknowledged {
+			// Not acknowledged
+		}
+	}
+
+	bulkRequest := e.client.Bulk()
+	for _, entry := range b.entries {
+		index := IndexEntry{Hash: entry.HashValue, Range: string(entry.RangeValue), Value: string(entry.Value)}
+		req := elastic.NewBulkIndexRequest().Index(indexName).Type(e.cfg.IndexType).Doc(index)
+		bulkRequest = bulkRequest.Add(req)
+	}
+
+	bulkResponse, err := bulkRequest.Do(ctx)
+	if err != nil {
+		return err
+	}
+
+	if bulkResponse != nil {
+
+	}
+	return nil
+}
+
+// readBatch represents a batch of rows read from ElasticSearch.
+type readBatch struct {
+	rangeValue []byte
+	value      []byte
+}
+
+func (r readBatch) Iterator() chunk.ReadBatchIterator {
+	return &elasticReadBatchIterator{
+		readBatch: r,
+	}
+}
+
+type elasticReadBatchIterator struct {
+	consumed bool
+	readBatch
+}
+
+func (r *elasticReadBatchIterator) Next() bool {
+	if r.consumed {
+		return false
+	}
+	r.consumed = true
+	return true
+}
+
+func (r *elasticReadBatchIterator) RangeValue() []byte {
+	return r.rangeValue
+}
+
+func (r *elasticReadBatchIterator) Value() []byte {
+	return r.value
+}
+
+func (e *esClient) QueryPages(ctx context.Context, queries []chunk.IndexQuery, callback func(chunk.IndexQuery, chunk.ReadBatch) (shouldContinue bool)) error {
+	return chunk_util.DoParallelQueries(ctx, e.query, queries, callback)
+}
+
+func (e *esClient) query(ctx context.Context, query chunk.IndexQuery, callback func(chunk.ReadBatch) (shouldContinue bool)) error {
+	var rangeQuery *elastic.RangeQuery
+	var valueTermQuery *elastic.TermQuery
+
+	level.Debug(util.Logger).Log("msg", fmt.Sprintf(
+		"hash [%s], rangeValuePrefix [%s], rangeValueStart [%s]", query.HashValue, query.RangeValuePrefix, query.RangeValueStart))
+	hashTermQuery := elastic.NewTermQuery("hash", query.HashValue)
+	switch {
+	case len(query.RangeValuePrefix) > 0 && query.ValueEqual == nil:
+		rangeQuery = elastic.NewRangeQuery("range").Gte(string(query.RangeValuePrefix)).
+			Lt(string(query.RangeValuePrefix) + null)
+
+	case len(query.RangeValuePrefix) > 0 && query.ValueEqual != nil:
+		rangeQuery = elastic.NewRangeQuery("range").Gte(string(query.RangeValuePrefix)).
+			Lt(string(query.RangeValuePrefix) + null)
+		valueTermQuery = elastic.NewTermQuery("value", query.ValueEqual)
+
+	case len(query.RangeValueStart) > 0 && query.ValueEqual == nil:
+		rangeQuery = elastic.NewRangeQuery("range").Gte(string(query.RangeValueStart))
+
+	case len(query.RangeValueStart) > 0 && query.ValueEqual != nil:
+		rangeQuery = elastic.NewRangeQuery("range").Gte(string(query.RangeValueStart))
+		valueTermQuery = elastic.NewTermQuery("value", query.ValueEqual)
+
+	case query.ValueEqual != nil:
+		valueTermQuery = elastic.NewTermQuery("value", query.ValueEqual)
+
+	case query.ValueEqual == nil:
+		break
+	}
+
+	exists, err := e.client.IndexExists(query.TableName).Do(ctx)
+	if err != nil {
+		// Handle error
+		fmt.Println(exists)
```

Comment:
This error handling doesn't log data through the go-kit library, but it also does not return the error upstream. this will lead to a panic later in the function